internal/components/extensions_test.go

```diff
@@ -267,6 +268,20 @@ func verifyExtensionLifecycle(t *testing.T, factory extension.Factory, getConfig
 	require.NoError(t, secondExt.Shutdown(ctx))
 }
 
+// verifyExtensionShutdown is used to test if an extension type can be shutdown without being started first.
+func verifyExtensionShutdown(t *testing.T, factory extension.Factory, getConfigFn getExtensionConfigFn) {
+	ctx := context.Background()
+	extCreateSet := extensiontest.NewNopCreateSettings()
+
+	if getConfigFn == nil {
+		getConfigFn = factory.CreateDefaultConfig
+	}
+
+	firstExt, _ := factory.CreateExtension(ctx, extCreateSet, getConfigFn())
+
+	require.NoError(t, firstExt.Shutdown(ctx))
```

Comment:
```suggestion 
	assert.NoError(t, firstExt.Shutdown(ctx))
```
Reason being is that went all components that fail to run, not just the first one.